### ISSUE TITLE
`if isinstance(instance, SuperClass):` loses type

### ISSUE THREAD

**Issue by jeremyBanks** [Microsoft/pyright#28]
Sort-of the reverse of #24: `InstanceOfTypeConstraint` constrains the subject's type to `None`/Never when it's checked against a superclass.

Maybe this should be a no-op instead? I'm not entirely sure, but mypy doesn't flag this as an error.

```python
class SuperClass:
    def __init__(self) -> None:
        self.property: None = None

class MyClass(SuperClass):
    pass

def f(instance: MyClass) -> None:
    if isinstance(instance, SuperClass):
        print(instance.property)
```

```none
$ npx pyright@1.0.4 server/src/tests/samples/isinstance.py
Finding source files
Found 1 source files
Analyzed 1 file in 0.981sec
/mnt/c/Users/_/pyright/server/src/tests/samples/isinstance.py
  'property' is not a known member of 'None' (10, 24)
1 error, 0 warnings
```

Equivalent example with `object` instead of an explicit superclass:

```python
class MyClass:
    def __init__(self) -> None:
        self.property: None = None

def f(instance: MyClass) -> None:
    if isinstance(instance, object):
        print(instance.property)
```

**Comment by erictraut** [comment]
The interactions between subclasses and superclasses was more complex than I had originally anticipated. I've reworked the logic, and I think it's now doing the right thing in all cases. The fix will be in the next version (1.0.5). Let me know if you see any further problems. And thanks for reporting the problem.

**Comment by jeremyBanks** [comment]
LGTM, thanks! I tried a few similar examples and pyright@master's results were consistent with MyPy.
